Remotion/Data/DomainObjects/Persistence/Rdbms/SchemaGeneration/RdbmsStructuredTypeDefinitionProvider.cs

```diff
@@ -0,0 +1,103 @@
+// This file is part of the re-motion Core Framework (www.re-motion.org)
+// Copyright (c) rubicon IT GmbH, www.rubicon.eu
+//
+// The re-motion Core Framework is free software; you can redistribute it
+// and/or modify it under the terms of the GNU Lesser General Public License
+// as published by the Free Software Foundation; either version 2.1 of the
+// License, or (at your option) any later version.
+//
+// re-motion is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with re-motion; if not, see http://www.gnu.org/licenses.
+//
+using System;
+using System.Collections.Generic;
+using System.Data;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Model;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.Model.Building;
+using Remotion.Data.DomainObjects.Persistence.Rdbms.SqlServer.Model.Building;
+using Remotion.Utilities;
+
+namespace Remotion.Data.DomainObjects.Persistence.Rdbms.SchemaGeneration
+{
+  /// <summary>
+  /// Provides an <see cref="IRdbmsStructuredTypeDefinition"/> for each simple type supported by <see cref="SqlStorageTypeInformationProvider"/>.
+  /// </summary>
+  public class RdbmsStructuredTypeDefinitionProvider : IRdbmsStructuredTypeDefinitionProvider
+  {
+    public IReadOnlyCollection<IRdbmsStructuredTypeDefinition> GetTypeDefinitions (RdbmsProviderDefinition rdbmsProviderDefinition)
+    {
+      ArgumentUtility.CheckNotNull(nameof(rdbmsProviderDefinition), rdbmsProviderDefinition);
+
+      var storageTypeInformationProvider = rdbmsProviderDefinition.Factory.CreateStorageTypeInformationProvider(rdbmsProviderDefinition);
```

Comment:
why did you inject the ProviderDef instead of the factory? it feels like it moved the complexity into the ctor but I don't see the benefit without more detail

Reply:
The factory's method `CreateStorageTypeInformationProvider()` needs the `rdbmsProviderDefinition` as parameter, so it would not help to only pass the factory without the `rdbmsProviderDefinition`.
I have already changed this in RM-9216, so this only pulls that aspect of the redesign to the current pull request in order to make the tests easier to write.